kobo/apps/trash_bin/tests/test_utils.py

```diff
@@ -592,3 +594,215 @@ def test_storage_does_not_change_on_archive_unarchive(self):
+        assert not AttachmentTrash.objects.filter(
+            attachment_id=self.attachment.id
+        ).exists()
+        assert not PeriodicTask.objects.filter(
+            pk=trash_obj.periodic_task_id
+        ).exists()
+
+        assert AuditLog.objects.filter(
+            app_label='logger',
+            model_name='attachment',
+            object_id=self.attachment.pk,
+            user=self.user,
+            action=AuditAction.PUT_BACK,
+            log_type=AuditType.ATTACHMENT_MANAGEMENT,
+        ).exists()
+
+    def test_trashing_attachment_deletes_file_but_preserves_db_object(self):
+        """
+        Test that moving an attachment to trash removes the file from storage
+        but not the attachment object itself
+        """
+        media_file = self.attachment.media_file
+        assert self.attachment.media_file.storage.exists(str(media_file))
+
+        trash_obj = self._move_attachment_to_trash(self.attachment, self.user)
+        empty_attachment(trash_obj.pk)
+
+        assert Attachment.all_objects.filter(pk=self.attachment.pk).exists()
+        assert not self.attachment.media_file.storage.exists(str(media_file))
+        assert not AttachmentTrash.objects.filter(
+            attachment_id=self.attachment.pk
+        ).exists()
+
+        assert AuditLog.objects.filter(
+            app_label='logger',
+            model_name='attachment',
+            object_id=self.attachment.pk,
+            user=self.user,
+            action=AuditAction.DELETE,
+            log_type=AuditType.ATTACHMENT_MANAGEMENT,
+            metadata={
+                'attachment_uid': self.attachment.uid,
+                'attachment_name': self.attachment.media_file_basename,
+                'instance__root_uuid': self.attachment.instance.root_uuid,
+            },
+        ).exists()
+
+    @data(
+        # Format: (task status, within grace period, is stuck, expected restart count)
+        # Trivial case: Task is pending and still within the grace period
+        (TrashStatus.PENDING, False, False, 0),
+        # Task is pending but the grace period has expired — it should be restarted
+        (TrashStatus.PENDING, True, False, 1),
+        # Task has started and is still running normally
+        (TrashStatus.IN_PROGRESS, True, False, 0),
+        # Task has started but is now stuck — it should be restarted
+        (TrashStatus.IN_PROGRESS, True, True, 1),
+        # Task failed but is not considered stuck — no restart needed
+        (TrashStatus.FAILED, True, False, 0),
+        # Task failed and is considered stuck — still no restart (handled differently)
+        (TrashStatus.FAILED, True, True, 0),
+    )
+    @unpack
+    def test_task_restarter(self, status, is_time_frozen, is_stuck, restart_count):
+        """
+        A freshly created task should not be restarted if it is in grace period.
+        """
+        if is_time_frozen:
+            frozen_time = '2024-12-10'
```

Comment:
A comment here about how freezing time determines whether something is or is not in the grace period would be helpful